lib/graph/tests/datum.cpp

```diff
@@ -269,3 +269,44 @@ TEST_CASE("Datum output detection")
     REQUIRE(out.count(y) == 1);
     delete g;
 }
+
+TEST_CASE("'self'")
+{
+    auto g = new Graph();
+    auto n = new Node("a", g);
+    auto x = new Datum("x", "123.0", &PyFloat_Type, n);
+    auto fromself = new Datum("fromself", "self.x", &PyFloat_Type, n);
+    auto expr = new Datum("expr", "self.x + 1", &PyFloat_Type, n);
+    // auto showfromself = new Datum("showfromself", Datum::SIGIL_OUTPUT + "fromself", &PyFloat_Type, n);
```

Comment:
I'd like to test that "output" datums get their value from a datum that uses "self". I don't know how to construct a datum for that.